clustering/consul

```diff
@@ -0,0 +1,465 @@
+version_added: "1.9"
+author: Steve Gargan (steve.gargan@gmail.com)
+options:
+    state:
+        description:
+          - register or deregister the consul service, defaults to present
+        required: true
+        choices: ['present', 'absent']
+    service_id:
+        description:
+          - the ID for the service, must be unique per node, defaults to the
+            service name
+        required: false
+    host:
+        description:
+          - host of the consul agent with which to register the service,
+            defaults to localhost
+        required: false
+    notes:
+        description:
+          - Notes to attach to check when registering it.
+    service_name:
+        desciption:
+          - Unique name for the service on a node, must be unique per node,
+            required if registering a service. May be ommitted if registering
+            a node level check
+        required: false
+    service_port:
+        description:
+          - the port on which the service is listening required for
+            registration of a service.
+        required: true
+    tags:
+        description:
+          - a list of tags that will be attached to the service registration.
+        required: false
+    script:
+        description:
+          - the script/command that will be run periodically to check the health
+            of the service
+        required: false
+    interval:
+        description:
+          - the interval at which the service check will be run. This is by
+            convention a number with a s or m to signify the units of seconds
+            or minutes. if none is supplied, m will be appended
+    check_id:
+        description:
+          - an ID for the service check, defaults to the check name, ignored if
+            part of service definition.
+    check_name:
+        description:
+          - a name for the service check, defaults to the check id. required if
+            standalone, ignored if part of service definition.
+"""
+
+EXAMPLES = '''
+  - name: register nginx service with the local consul agent
+    consul:
+      name: nginx
+      port: 80
+
+  - name: register nginx service with curl check
+    consul:
+      name: nginx
+      port: 80
+      script: "curl http://localhost"
+      interval: 60s
+
+  - name: register nginx with some service tags
+    consul:
+      name: nginx
+      port: 80
+      tags:
+        - prod
+        - webservers
+
+  - name: remove nginx service
+    consul:
+      name: nginx
+      state: absent
+
+  - name: create a node level check to test disk usage
+    consul:
+      check_name: Disk usage
+      check_id: disk_usage
+      script: "/opt/disk_usage.py"
+      interval: 5m
+
+'''
+
+import sys
+import urllib2
+
+try:
+    import json
+except ImportError:
+    import simplejson as json
+
+try:
+    import consul
+except ImportError, e:
+    print "failed=True msg='python-consul required for this module. "\
```

Comment:
This took the lead from other modules importing third party libraries and It does fail if the library is not present. I'll look at importing the library in the scope of the module and use the fail_json instead.